# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpBackgroundChannelParent.h
# Commit: 73b51a139c91
# Full Hash: 73b51a139c915c7e4aa7b9bff58769bd21cfcd78
# Author: Junior Hsu <juhsu@mozilla.com>
# Date: 2020-06-24 09:31:07
# Regressor Bug: 1633935
# File Overlap Count: 1
# Description:
#   Bug 1633935 - P5 allow OnProgress/OnStatus going thru pBg IPC, r=mayhemer,necko-reviewers
#   
#   Depends on D76971
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D76972
# ==============================================================================

diff -r 2a9873ad6856 -r 73b51a139c91 netwerk/protocol/http/HttpBackgroundChannelParent.h
--- a/netwerk/protocol/http/HttpBackgroundChannelParent.h	Tue Jun 23 16:56:53 2020 +0000
+++ b/netwerk/protocol/http/HttpBackgroundChannelParent.h	Tue Jun 23 16:57:10 2020 +0000
@@ -57,6 +57,12 @@
                      const nsHttpHeaderArray& aResponseTrailers,
                      const nsTArray<ConsoleReportCollected>& aConsoleReports);
 
+  // To send OnProgress message over background channel.
+  bool OnProgress(const int64_t aProgress, const int64_t aProgressMax);
+
+  // To send OnStatus message over background channel.
+  bool OnStatus(const nsresult aStatus);
+
   // To send FlushedForDiversion and DivertMessages messages
   // over background channel.
   bool OnDiversion();